### PR TITLE
feat: Update the go feature server from Expedia code repo.

### PR DESCRIPTION
**Pull request by shuchu**
# What this PR does / why we need it:

Update the improved Go feature server code from ExpediaGroup/Feast to our code repo.

# Which issue(s) this PR fixes:

### CODE REVIEW

**Comment by shuchu** [comment]
cc: @EXPEbdodla

**Comment by shuchu** [comment]
[Update 10-22-2024] Now it can compile, build and build docker image. All configures in the Makefile under the "#Go SDK & embedded" are working, except the "install-go-ci-dependencies".  

The current "install-go-ci-dependencies" is obsoleted. We need to fix it when we setup the integration test for Go feature server. 

!!Warning, the current version my not function correctly. need more testing and investigating.

**Comment by shuchu** [comment]
[Update 10-31-2024] 

1.  "make test-go" works now
2.  DataDog related observability instrumentation code were commented out but not removed. I keep these code and plan to use OTEL to replace them.

File: go/internal/feast/registry/http.go
Comment (EXPEbdodla) [inline go/internal/feast/registry/http.go:1]: This needs to be switched to remote registry. We built http registry to overcome the password sharing for SQL Registry. HTTP Registry which is FastAPI based is not contributed to Feast. We are also planning to leverage Remote registry and move over from HTTP one.

Comment (shuchu) [reply go/internal/feast/registry/http.go:1]: Thank you for the info, @EXPEBdodla ! Let me see try to resolve this.

**Comment by shuchu** [comment]
[Update 11/04/2024] 

1. The HTTP Registry code is removed as it depends on an external HTTP server as @EXPEbdodla pointed out.
2.  The default Registry is File based one now.

File: go/internal/feast/featurestore.go
Comment (shuchu) [inline go/internal/feast/featurestore.go:67]: Is this an external Transformation server? I guess we need to remove this and related code? @EXPEbdodla

Comment (EXPEbdodla) [reply go/internal/feast/featurestore.go:67]: Initially transformations are running with GOPY bindings. That's not working at scale. So we implemented it similar to Java Feature Server.

Comment (shuchu) [reply go/internal/feast/featurestore.go:67]: Thank you, @EXPEbdodla  I guess I need to work on this part.

File: go/internal/feast/registry/http.go
Comment (shuchu) [reply go/internal/feast/registry/http.go]: I choose an "easy" fix for this.

File: go/internal/feast/featurestore.go
Comment (shuchu) [reply go/internal/feast/featurestore.go]: I chose an "easy" fix for this. :crying_cat_face:

File: go/internal/feast/featurestore.go
Comment (HaoXuAI) [inline go/internal/feast/featurestore.go:318]: I think these can be removed or updated as datadog is not part of feast

File: go/internal/feast/onlinestore/redisonlinestore.go
Comment (HaoXuAI) [inline go/internal/feast/onlinestore/redisonlinestore.go:109]: similar here

File: go/README.md
Comment (HaoXuAI) [inline go/README.md:1]: Any reason we are removing the document here?

File: go/internal/feast/featurestore.go
Comment (EXPEbdodla) [reply go/internal/feast/featurestore.go]: Yes. We should get the transformation server endpoint from feature_store.yaml.

File: go/internal/feast/featurestore.go
Comment (EXPEbdodla) [reply go/internal/feast/featurestore.go:318]: Yea. Datadog needs to be replaced with OTEL.

File: go/README.md
Comment (EXPEbdodla) [reply go/README.md:1]: That piece of code is not needed if we remove GOPY bindings.

File: go/internal/feast/featurestore.go
Comment (shuchu) [reply go/internal/feast/featurestore.go:318]: yes, I will replace them with OTEL after this PR get merged.

**Comment by shuchu** [comment]
[Update 11/24/2024]

1.  use the feature_store.yaml file as the source of defining Python transformation service.  Example:
```
# feature_store.yaml file
feature_store:
   transformation_service_endpoint: "localhost:50051"
```

@EXPEbdodla @feast-dev/reviewers-and-approvers

File: go/internal/feast/featurestore.go
Comment (shuchu) [reply go/internal/feast/featurestore.go]: updated. :)

File: Makefile
Comment (franciscojavierarceo) [inline Makefile:550]: this is all commented out, do we need it?

File: Makefile
Comment (franciscojavierarceo) [inline Makefile:554]: is this make command correct?

Comment (franciscojavierarceo) [inline Makefile:554]: the inline seems to be at odds with the rest of the Makefile format, could you adjust it?

File: Makefile
Comment (franciscojavierarceo) [inline Makefile:566]: same here

File: go/internal/feast/onlinestore/sqliteonlinestore.go
Comment (franciscojavierarceo) [inline go/internal/feast/onlinestore/sqliteonlinestore.go:44]: was this just a lint format change?

File: go/internal/feast/registry/registry.go
Comment (franciscojavierarceo) [inline go/internal/feast/registry/registry.go:48]: This seems like a pretty large change in time proportional to the number of seconds, no? Was this on purpose?

File: go/internal/feast/transformation/transformation.go
Comment (franciscojavierarceo) [inline go/internal/feast/transformation/transformation.go:94]: rip

File: Makefile
Comment (shuchu) [reply Makefile:550]: commented out as I'm not sure if this setting is good/ready for CI test.

File: go/internal/feast/registry/registry.go
Comment (shuchu) [reply go/internal/feast/registry/registry.go:48]: This one may need @EXPEbdodla 's help to explain.

File: Makefile
Comment (shuchu) [reply Makefile]: done

File: Makefile
Comment (shuchu) [reply Makefile]: done

File: Makefile
Comment (shuchu) [reply Makefile]: done

File: go/internal/feast/onlinestore/redisonlinestore.go
Comment (shuchu) [reply go/internal/feast/onlinestore/redisonlinestore.go]: done

File: go/internal/feast/onlinestore/sqliteonlinestore.go
Comment (shuchu) [reply go/internal/feast/onlinestore/sqliteonlinestore.go]: fixed.

**Review by franciscojavierarceo — APPROVED** [review]
🚀 🚀

**Pull request merged**
